### PR TITLE
test: add route with uri or uris ( E2E )

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E test
- [ ] New feature provided
- [ ] Improve performance

- Related issues
https://github.com/apache/apisix-dashboard/issues/632

### CODE REVIEW

File: api/test/e2e/route_with_uri_uris_test.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestRoute_with_invalid_uri_uris(t *testing.T) {
+	tests := []HttpTestCase{
+		{
+			caseDesc: "add route with invalid uri",
```
Comment (membphis) [inline api/test/e2e/route_with_uri_uris_test.go:27]: please confirm you are using an invalid `uri` for this case

'/hello_notexist' seems to be a valid uri value.

File: api/test/e2e/route_with_uri_uris_test.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestRoute_with_invalid_uri_uris(t *testing.T) {
```
Comment (membphis) [inline api/test/e2e/route_with_uri_uris_test.go:24]: This test case is useless, I think we can delete it.

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_uri_uris_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+							"port": 1980,
+							"weight": 1
+						}]
+					}
+				}`,
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+		},
+		{
+			caseDesc:     "verify route",
```
Comment (membphis) [inline api/test/e2e/route_with_uri_uris_test.go:84]: Need a better name.

File: api/test/e2e/route_with_uri_uris_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "hello world",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc:     "verify route",
```
Comment (membphis) [inline api/test/e2e/route_with_uri_uris_test.go:94]: ditto

**Review by membphis — APPROVED** [review]
Look good for me, except minor things.

**Comment by nic-chen** [comment]
should aways verify before create resource and after delete resource.

**Review by gxthrj — APPROVED** [review]

**Comment by membphis** [comment]
> should aways verify before create resource and after delete resource.

For the first step, we do not have to do this.

Every test case should do clean all of the temporary data by itself.

File: api/test/e2e/route_with_uri_uris_test.go (hunk 2 above)
Comment (idbeta) [reply api/test/e2e/route_with_uri_uris_test.go]: fixed.

File: api/test/e2e/route_with_uri_uris_test.go (hunk 3 above)
Comment (idbeta) [reply api/test/e2e/route_with_uri_uris_test.go]: fixed.

**Pull request merged**